Embedd hypervisor-crd, inital switch to hypervisor-crd based controllers

notandy opened the pull request:
Improved visibility of the operator status reflected in hypervisor crd:

```
$ k get hv -o wide
NAME            STATE   LIFECYCLE   HIGH AVAILABILITY   VERSION               INSTANCES   HARDWARE               KERNEL          ONBOARDING   SERVICE ID                             HYPERVISOR ID                          AGE
node004-bb088           true        true                Garden Linux 1976.0   0           ProLiant DL380 Gen11   6.12.43-amd64   completed    4a5be510-1ce8-499b-881f-ce643117e092   b5204eaf-d521-5d40-b88f-565036895ea4   6h56m
node005-bb086           true        true                Garden Linux 1977.0   1           PowerEdge R860         6.12.43-amd64   testing      de058d04-7929-4e6a-8511-d8955dee01cd   18870cb0-9910-5486-b31e-4811f4895236   5h18m
```

notandy committed:
Embedd hypervisor-crd, inital switch to hypervisor-crd based controllers

mchristianl reviewed (dismissed):
I have not yet reviewed the content of your PR but here are some minor things already

notandy committed:
Transfer labels from node to hypervisor object

also interprets the lifecycle label now again for persistence, added a
test for hypervisor-controller

fwiesel reviewed (approved):
The `retry.RetryOnConflict` should now be superfluous, but I wouldn't say that is a reason to stop merging it as it is.
